custom/icds_reports/data_pull/exporter.py

```diff
@@ -30,7 +30,7 @@ def __init__(self, slug_or_file, db_alias, month, location_id):
     @prevent_parallel_execution(DATA_PULL_CACHE_KEY)
     def export(self):
         zip_file_name = "%s-DataPull.zip" % self.data_pull_obj.name
-        with zipfile.ZipFile(zip_file_name, mode='a') as z:
+        with zipfile.ZipFile(zip_file_name, mode='w') as z:
```

Comment:
Why is this changed?

Reply:
since we are creating a new file "w" would the write thing to avoid appending to an existing file?

Reply:
how was being handled previously?

Reply:
It didn't really cause any issue earlier because any request was not repeated but good to keep it new each time.
Also zip file name i think has the timestamp as well so reusing a file won't happen but have a new file sounds like the right thing to do.

Reply:
or you are saying previously, we were using it once so it was not appending and if we were running it twice we used to delete the old one?

Reply:
Actually we have never run this more than once. Plus we remove the file after use
https://github.com/dimagi/commcare-hq/blob/3a8e462914827c351efe10a1d36e8cbfc67cf3e5/custom/icds/tasks/data_pulls.py#L57